cfspopcon/formulas/separatrix_operational_space/auxillaries.py

```diff
@@ -0,0 +1,92 @@
+"""Parameters which are used to calculate the separatrix operational space, but which aren't directly outputted."""
+import numpy as np
+
+from ...unit_handling import Quantity, Unitfull, convert_units, ureg
+
+
+def calc_cylindrical_edge_safety_factor(
+    major_radius: Unitfull,
+    minor_radius: Unitfull,
+    elongation: Unitfull,
+    triangularity: Unitfull,
+    magnetic_field_strength: Unitfull,
+    plasma_current: Unitfull,
+) -> Unitfull:
+    """Calculate the edge safety factor, following the formula used in the SepOS paper.
+
+    Equation K.6 from :cite:`Eich_2021`
+
+    Should use kappa_95 and delta_95 values.
+
+    Gives a slightly different result to our standard q_star calculation.
+    """
+    shaping_correction = np.sqrt((1.0 + elongation**2 * (1.0 + 2.0 * triangularity**2 - 1.2 * triangularity**3)) / 2.0)
```

Comment:
This term seems common. It's also in calc_f_shaping so could be worth factoring out and sharing in these places.

https://github.com/cfs-energy/cfspopcon/pull/12#discussion_r1465410375